server/scores.js

```diff
@@ -15,16 +21,54 @@ const connectionPromise = MongoClient
   .connect(mongoUrl, { promiseLibrary: Promise, useNewUrlParser: true })
   .then(client => client.db().collection('scores'))
 
+function _validateScore (score) {
+  let retError = { 'status': 'ok', 'errors': '' }
+
+  if (typeof score.teamNumber !== 'number' || score.match == null || score.score == null) {
+    retError.status = 'loud-fail'
+    if (typeof score.teamNumber !== 'number') { retError.errors += 'team number ' }
+    if (score.score == null) { retError.errors += 'score ' }
+    if (score.match == null) { retError.errors += 'match ' }
+  }
+
+  if (score.signature.isEmpty) {
+    retError.errors += 'signature '
+    retError.status = retError.status === 'loud-fail' ? 'loud-fail' : 'silent-fail'
+  }
+
+  return retError
+}
+
 const adminAction = authroizationMiddlware(['admin', 'scorekeeper', 'development'])
 
 router.post('/create', (req, res) => {
-  connectionPromise
-    .then(scoringCollection => scoringCollection.save(req.body))
-    .then(() => res.status(201).send())
-    .catch(err => {
-      req.logger.error(err.message)
-      res.status(500).send('A problem occoured while trying to save score.')
-    })
+  const scoreValidation = _validateScore(req.body)
+
+  if (scoreValidation.status !== 'ok') {
+    req.logger.error('Invalid score, missing ' + scoreValidation.errors + '. ' + scoreValidation.status + '.')
+  }
+
+  if (scoreValidation.status !== 'loud-fail') {
+    connectionPromise
+      .then(scoringCollection => {
+        scoringCollection.save(req.body)

```

Comment:
You're still not sanitizing the score. The `_validateScore` function would return a valid score, with only the allowed fields, and that score should be saved.
If it turns the score, it cannot return a validation status, so you should use the exception mechanism (i.e. throw a custom exception) that declares the score to be invalid, and that would let the routing method know to return a 422.

Reply:
Is the existing if on line 51 which checks whether or not there is a loud fail precisely what you want here? It also indeed does throw a 422 if there is a loud fail.

Reply:
The functionality here is good, and it is performed by the code. This comment is regarding the method of implementing this functionality

Reply:
Notice I updated my comment to be clearer